src/components/Header/HeaderParts/SearchBox.tsx

```diff
@@ -12,27 +23,350 @@ interface SearchBoxProps {
 }
 
 export default function SearchBox({ isBlack }: SearchBoxProps) {
-  const [searchValue, setSearchValue] = useState<string>('');
+  const router = useRouter();
+  const pathName = usePathname();
+
+  const { data: SearchSuggestionData } = useQuery<string[]>({
+    queryFn: getSearchSuggestion,
+    queryKey: ['searchSuggestionData'],
+  });
+
+  const wrapperRef = useRef<HTMLDivElement>(null);
+  const inputRef = useRef<HTMLInputElement>(null);
+  const suggestRef = useRef<HTMLDivElement>(null);
+  const timerRef = useRef<NodeJS.Timeout | null>(null);
+
+  const [isOpen, setIsOpen] = useState(false);
+  const [isRender, setIsRender] = useState(false);
+
+  const [searchKeyword, setSearchKeyword] = useState('');
+  const [originKeyword, setOriginKeyword] = useState('');
+  const [isFocus, setIsFocus] = useState(false);
+  const [autoListType, setAutoListType] = useState<'history' | 'suggestion'>('history');
+  const [focusIndex, setFocusIndex] = useState(-1);
+
+  const [keywordHistory, setKeywordHistory] = useState<string[]>([]);
+  const [keywordSuggestion, setKeywordSuggestion] = useState<SuggestionDataType[]>([]);
+
+  /* 
+    외부 클릭 시 focus 해제
+    자동 완성 리스트 안 뜨도록 설정 => 클릭한 곳이 suggestRef 내부일 경우 제외
+   */
+  useOutsideClick(inputRef, (e?: MouseEvent) => {
+    if (e && suggestRef.current && !suggestRef.current.contains(e.target as Node)) {
+      setIsFocus(false);
+    }
+  });
+
+  /* 
+    헤더 검색버튼 눌렀을 때 함수
+    검색 컴포넌트 열 때는 isRender도 같이 true로 바로 바뀌도록 설정/ 닫혔을 때는 비활성
+    => 애니메이션이 끝난 후에 isRender 값이 false로 바뀌도록 하기 위해
+  */
+  const handleClickButton = () => {
+    setIsOpen((prev) => {
+      if (!prev) {
+        setIsRender(true);
+      }
+      return !prev;
+    });
+  };
+
+  /* 
+    검색 컴포넌트 unmount 요청 시(isOpen값이 false로 바뀔 때)에 animation이 발생하고 다 끝난 후에 unmount되도록 하기 위한 코드. 
+    애니메이션이 종료된 후 isOpen이 false 일 때(사용자가 닫는 조건을 발동시켰을 때 - 외부 클릭 또는 헤더 검색 버튼을 누른 경우 ) unmount 되도록 설정
+    참고로 애니메이션은 처음에 isOpen이 true로 변경될 때 또는 false로 변경될 때 발동되며(slide-in, slide-out),
+    닫히는 경우, isOpen이 false로 변경되고, 애니메이션이 일어난 후, unmount되어야 함(isRender가 false)
+  */
+  const handleAnimationEnd = () => {
+    if (!isOpen) {
+      setIsRender(false);
+    }
+  };
+
+  /* 
+    회색 배경 부분 클릭 했을 경우 검색 컴포넌트 닫히도록 설정
+    만약 헤더 클릭 했을 때도 닫히도록 하려면 그냥 useOutsideClick으로 하면 됨.
+  */
+  const handleClickOutside = (e: React.MouseEvent<HTMLDivElement>) => {
+    if (wrapperRef.current && !wrapperRef.current.contains(e.target as Node)) {
+      handleClickButton();
+    }
+  };
```

Comment:
```suggestion
  const handleClickOutside = (e: MouseEvent<HTMLDivElement>) => {
    if (wrapperRef.current && !wrapperRef.current.contains(e.target as Node)) {
      handleClickButton();
    }
  };
```
import 해서 사용하시는게 저희 컨벤션에 더 맞는듯 합니다

Reply:
이거 React.MouseEvent하고 그냥 자바스크립트의 mouseEvent하고 구분이 안되어서 이런 식으로 사용했습니다(위의 useOutsideClick 훅에서 일반 mouseEvent 사용)

Reply:
구문이 안된다면 as로 변경하거나 type지정을 잘못한게 아닐까요?
이부분은 저도 잘 모르겠네요...하지만 컨벤션만 생각했을때에는 좋지않아보이는건 확실한것 같습니다